src/main/java/fi/aalto/cs/apluscourses/presentation/exercise/ExerciseViewModel.java

```diff
@@ -18,6 +18,38 @@ public String getPresentableName() {
     return APlusLocalizationUtil.getEnglishName(getModel().getName());
   }
 
+  /**
+   * Returns {@code true} if the exercise is submittable from the plugin, {@code false} otherwise.
+   */
+  public boolean isSubmittable() {
+    String presentableName = getPresentableName();
+    return presentableName.length() > "Assignment xx (".length()
+        && !"Assignment 1 (Piazza)".equals(presentableName);
+  }
+
+  public enum Status {
+    NO_SUBMISSIONS,
+    NO_POINTS,
+    PARTIAL_POINTS,
+    FULL_POINTS,
+  }
+
+  /**
+   * Returns a {@link Status} that indicates the points status of this exercise.
+   */
+  public Status getStatus() {
```

Comment:
This method and `isSubmittable()` would make more sense in model. They do not only give information of how the state of the exercise is rendered for the user (view model things) but of the exercise object itself.

Reply:
I initially wanted to put `isSubmittable` in the model class, but I decided against it for the following reasons:

1. The `isSubmittable` method uses a hacky solution, and therefore trying to submit a disabled exercise is still possible, just in case that `isSubmittable` produces an incorrect result.
2. This hacky method used to determine whether to show the exercise as enabled or not, may cause confusion with the `isSubmittable` method from `SubmissionInfo`, which actually looks at the submission form files.

Any opinions?

`getStatus` could definitely go in the model class, although I wonder if there is any benefit from having a separate `Status` enum, when all of the values can be figured out from the values from the exercise...

Reply:
Good reasoning, let's keep hackiness out of the model. When it comes to `getStatus`, I think both of the viewpoints are valid. We can think the status resolving as a part of the business logic or just part of the presentation logic.

So no need to change.